.circleci/config.yml

```diff
@@ -1,73 +1,110 @@
-
-      - save_cache:
-          name: Save git cache
-          key: v1-20180524-git
-          paths:
-            - ".git"
-
-      - run:
-          name: Create Directories for Results and Artifacts
-          command: |
-            mkdir -p $CIRCLE_ARTIFACTS
-            mkdir -p $CIRCLE_TEST_REPORTS
-
-      - restore_cache:
-          name: "Restoring npm cache"
-          keys:
-             - v1-npmcache-{{ checksum ".nvmrc" }}-{{ checksum "npm-shrinkwrap.json" }}
-             - v1-npmcache-{{ checksum ".nvmrc" }}
-
+      # npm dependencies
+      - restore_cache: *restore-npm-cache
       - run: npm ci
+      - save_cache: *save-npm-cache
 
-      - save_cache:
-          name: "Saving npm cache"
-          key: v1-npmcache-{{ checksum ".nvmrc" }}-{{ checksum "npm-shrinkwrap.json" }}
-          paths:
-            - ~/.npm
+      # folders to collect results
+      - run: *setup-results-and-artifacts
+      ###################
+      # End Basic Setup #
+      ###################
 
       - run:
           name: Build calypso-strings.pot
           command: |
             if [ "$CIRCLE_NODE_INDEX" == "0" ]; then
-              npm run translate; mkdir -p $CIRCLE_ARTIFACTS/translate
-              mv calypso-strings.pot $CIRCLE_ARTIFACTS/translate
+              npm run translate
+              mv calypso-strings.pot "$CIRCLE_ARTIFACTS/translate"
             fi
 
       - run:
           name: Build New Strings .pot
           command: |
             if [ "$CIRCLE_NODE_INDEX" == "0" ]; then
               git clone https://github.com/Automattic/gp-localci-client.git
-              bash gp-localci-client/generate-new-strings-pot.sh $CIRCLE_BRANCH $CIRCLE_SHA1 $CIRCLE_ARTIFACTS/translate
+              bash gp-localci-client/generate-new-strings-pot.sh "$CIRCLE_BRANCH" "$CIRCLE_SHA1" "$CIRCLE_ARTIFACTS/translate"
               rm -rf gp-localci-client
             fi
 
       - run:
           name: Run Integration Tests
           command: |
-            bin/run-integration $(circleci tests glob "bin/**/integration/*.js" "client/**/integration/*.js" "server/**/integration/*.js" | circleci tests split --split-by=timings)
+            if [[ "$RUN_ARGS" != "--nightly" ]]; then
+              echo "Regular build: skipping tests."
+              exit 0
+            fi
+
+            JEST_JUNIT_OUTPUT="$CIRCLE_TEST_REPORTS/integration/test-results-integration.xml" npx --no-install \
+              jest                                         \
+                --ci                                       \
+                --reporters=default                        \
+                --reporters=jest-junit                     \
+                --config=test/integration/jest.config.json \
+                --maxWorkers=2                             \
+                $(circleci tests glob "bin/**/integration/*.js" "client/**/integration/*.js" "server/**/integration/*.js" | circleci tests split --split-by=timings)
```

Comment:
for things like this how would having some scripts in a `Makefile` impact the reachability of the tests? does inlining them here make it harder to run the integration tests locally or is it totally independent of that?

Reply:
> for things like this how would having some scripts in a `Makefile` impact the reachability of the tests?

I was hoping to hear some opinions about the inlining vs. npm script vs. bins vs. ?

I'm not very proficient in `make`, but all we need to do from this config file is to execute something from the shell and read its exit status (0/non-0).

I like the inlining because it gets rid of CI-related clutter around the repo and makes it easy to see what's happening right in the CI config.

> does inlining them here make it harder to run the integration tests locally or is it totally independent of that?

The tests continue to be available locally as `npm run test-{client,server,integration}`. They had `npm run test-{client,server,integration}:ci` variations that I don't think anyone would run locally, which I've removed in this PR.